diff -Naur a/configs/apache/DEADJOE b/configs/apache/DEADJOE
--- a/configs/apache/DEADJOE	1970-01-01 01:00:00.000000000 +0100
+++ b/configs/apache/DEADJOE	2007-11-15 21:34:44.000000000 +0100
@@ -0,0 +1,40 @@
+
+*** Modified files in JOE when it aborted on Thu Nov 15 21:34:44 2007
+*** JOE was aborted by signal 1
+
+*** File '(Unnamed)'
+/srv/htdocs/own-domains/
+/srv/htdocs/sponsoring/
+/srv/htdocs/kunden/
+CS_USER
+#Serhildan-yonetim
+/srv/www
+PF-Gaming.de
+PenguinFriends
+eth0
+gutsy
+
+*** File '(Unnamed)'
+streams
+globals
+lastella
+iwotronic
+Band
+long
+globals
+max
+file
+ROOT_APACHE
+
+*** File '(Unnamed)'
+header.html
+proftpd.conf
+httpd.conf
+proftpd/proftpd.conf
+/etc/apache2/vhosts.d/vh
+Makefile
+vhcs2.conf
+apache/Makefile
+vhcs2.conf
+apache/Makefile
+httpd.conf
diff -Naur a/configs/dists/suse/init.d/vhcs2_daemon b/configs/dists/suse/init.d/vhcs2_daemon
--- a/configs/dists/suse/init.d/vhcs2_daemon	2005-12-31 12:12:44.000000000 +0100
+++ b/configs/dists/suse/init.d/vhcs2_daemon	2007-11-01 22:29:28.000000000 +0100
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /bin/bash
 #
 # skeleton	example file to build /etc/init.d/ scripts.
 #		This file should be used to construct scripts for /etc/init.d.
diff -Naur a/configs/init.d/vhcs2_daemon b/configs/init.d/vhcs2_daemon
--- a/configs/init.d/vhcs2_daemon	2005-12-31 12:12:23.000000000 +0100
+++ b/configs/init.d/vhcs2_daemon	2007-11-01 22:12:54.000000000 +0100
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /bin/bash
 #
 # skeleton	example file to build /etc/init.d/ scripts.
 #		This file should be used to construct scripts for /etc/init.d.
diff -Naur a/configs/postfix/main.cf b/configs/postfix/main.cf
--- a/configs/postfix/main.cf	2005-12-31 12:12:58.000000000 +0100
+++ b/configs/postfix/main.cf	2007-11-01 14:48:58.000000000 +0100
@@ -77,10 +77,16 @@
 
 smtpd_sasl_security_options = noanonymous
 
-smtpd_sasl_local_domain = vhcs.net
+smtpd_sasl_local_domain = $myhostname
 
 broken_sasl_auth_clients = yes
 
+smtpd_sender_login_maps = hash:/etc/postfix/vhcs2/aliases
+
+smtpd_sender_restrictions = 
+   permit_mynetworks,
+   reject_sender_login_mismatch
+
 smtpd_recipient_restrictions =
    permit_sasl_authenticated,
    permit_mynetworks,
diff -Naur a/configs/proftpd/proftpd.conf b/configs/proftpd/proftpd.conf
--- a/configs/proftpd/proftpd.conf	2005-12-31 12:13:02.000000000 +0100
+++ b/configs/proftpd/proftpd.conf	2008-01-29 18:59:07.000000000 +0100
@@ -13,12 +13,20 @@
 ShowSymlinks			on
 AllowOverwrite			on
 
+SystemLog           		/var/log/proftpd.log
+
 LogFormat 			traff "%b %u"
 
 TimeoutNoTransfer		600
 TimeoutStalled			600
 TimeoutIdle			1200
 
+ServerIdent                    	Off
+IdentLookups			Off
+UseReverseDNS         		Off
+
+UseIPv6				Off
+
 DisplayLogin                    welcome.msg
 DisplayFirstChdir               .message
 
@@ -35,6 +43,8 @@
 
 Port				21
 
+Include 			/etc/proftpd/modules.conf
+
 # To prevent DoS attacks, set the maximum number of child processes
 # to 30.  If you need to allow more than 30 concurrent connections
 # at once, simply increase this value.  Note that this ONLY works
@@ -46,7 +56,7 @@
 
 # Set the user and group that the server normally runs at.
 
-User				nobody
+User				proftpd
 Group				nogroup
 
 # Normally, we want files to be overwriteable.
@@ -75,6 +85,8 @@
 	DelayEngine off
 </IfModule>
 
+SQLBackend mysql
+
 #
 # VHCS2 Managment;
 #
@@ -101,4 +113,4 @@
 
 QuotaLock /var/run/proftpd/tally.lock
 QuotaLimitTable sql:/get-quota-limit
-QuotaTallyTable sql:/get-quota-tally/update-quota-tally/insert-quota-tally
+QuotaTallyTable sql:/get-quota-tally/update-quota-tally/insert-quota-tally
\ No newline at end of file
diff -Naur a/configs/vhcs2.conf b/configs/vhcs2.conf
--- a/configs/vhcs2.conf	2006-01-03 20:56:32.000000000 +0100
+++ b/configs/vhcs2.conf	2008-04-14 17:49:53.000000000 +0200
@@ -329,7 +329,7 @@
 
 HTACCESS_GROUPS_FILE_NAME = .htgroup
 
-HTPASSWD_CMD = /usr/bin/htpasswd2
+HTPASSWD_CMD = /usr/bin/htpasswd
 
 #
 # backup management (16.05.2005).
diff -Naur a/engine/setup/vhcs2-setup b/engine/setup/vhcs2-setup
--- a/engine/setup/vhcs2-setup	2006-01-02 18:53:19.000000000 +0100
+++ b/engine/setup/vhcs2-setup	2007-11-01 22:16:40.000000000 +0100
@@ -641,13 +641,13 @@
     
     my ($rs, $rdata) = (undef, undef);
 
-    my $cmd = "/bin/sh $main::cfg{'ROOT_DIR'}/engine/setup/set-gui-permissions.sh";
+    my $cmd = "/bin/bash $main::cfg{'ROOT_DIR'}/engine/setup/set-gui-permissions.sh";
 
     $rs = sys_command($cmd);
 
     return $rs if ($rs != 0);
 
-	$cmd = "/bin/sh $main::cfg{'ROOT_DIR'}/engine/setup/set-engine-permissions.sh";
+	$cmd = "/bin/bash $main::cfg{'ROOT_DIR'}/engine/setup/set-engine-permissions.sh";
     
     $rs = sys_command($cmd);
 
diff -Naur a/gui/admin/view_ticket.php b/gui/admin/view_ticket.php
--- a/gui/admin/view_ticket.php	2005-12-31 12:10:47.000000000 +0100
+++ b/gui/admin/view_ticket.php	2008-03-27 14:45:27.000000000 +0100
@@ -281,7 +281,7 @@
     $ticket_date = time();
 
     $subj = $_POST['subject'];
-
+    htmlspecialchars($_POST["user_message"], ENT_QUOTES, "UTF-8");
     $user_message = preg_replace("/\n/", "<br>", $_POST["user_message"]);
 
 	$ticket_status = 1;
@@ -336,7 +336,7 @@
                                          $urgency,
                                          $ticket_date,
                                          htmlspecialchars($subj, ENT_QUOTES, "UTF-8"),
-                                         htmlspecialchars($user_message, ENT_QUOTES, "UTF-8")));
+                                         $user_message));
 
 	set_page_message(tr('Message was send!'));
 
diff -Naur a/gui/client/new_ticket.php b/gui/client/new_ticket.php
--- a/gui/client/new_ticket.php	2005-12-31 12:11:22.000000000 +0100
+++ b/gui/client/new_ticket.php	2008-03-27 14:47:22.000000000 +0100
@@ -47,6 +47,7 @@
   $ticket_date = time();
   $urgency = $_POST['urgency'];
   $subj = $_POST['subj'];
+  htmlspecialchars($_POST["user_message"], ENT_QUOTES, "UTF-8");
   $user_message = preg_replace("/\n/", "<br>", $_POST["user_message"]);
 	$ticket_status = 1;
 	$ticket_reply = 0;
@@ -69,7 +70,7 @@
                                        $urgency,
                                        $ticket_date,
                                        htmlspecialchars($subj, ENT_QUOTES, "UTF-8"),
-                                       htmlspecialchars($user_message, ENT_QUOTES, "UTF-8")));
+                                       $user_message));
 
   send_tickets_msg($reseller_id,$user_id,$subj);
   set_page_message(tr('Your message was sent!'));
diff -Naur a/gui/client/view_ticket.php b/gui/client/view_ticket.php
--- a/gui/client/view_ticket.php	2005-12-31 12:11:25.000000000 +0100
+++ b/gui/client/view_ticket.php	2008-03-27 14:47:54.000000000 +0100
@@ -209,6 +209,7 @@
 
   $ticket_date = time();
   $subj = $_POST['subject'];
+  htmlspecialchars($_POST["user_message"], ENT_QUOTES, "UTF-8");
   $user_message = preg_replace("/\n/", "<br>", $_POST["user_message"]);
   $ticket_status = 4;
 	$ticket_reply = $_GET['ticket_id'];
@@ -237,7 +238,7 @@
                                        $urgency,
                                        $ticket_date,
                                        htmlspecialchars($subj, ENT_QUOTES, "UTF-8"),
-                                       htmlspecialchars($user_message, ENT_QUOTES, "UTF-8")));
+                                       $user_message));
 
   set_page_message(tr('Message was send!'));
 
diff -Naur a/gui/include/login.php b/gui/include/login.php
--- a/gui/include/login.php	2005-12-31 12:09:05.000000000 +0100
+++ b/gui/include/login.php	2006-02-09 21:55:10.000000000 +0100
@@ -230,14 +230,20 @@
 
             header("Location: ../admin/manage_users.php");
 
+	    die();
+
         } else if ($_SESSION['user_type'] == 'reseller') {
 
             header("Location: ../reseller/index.php");
 
+	    die();
+
         } else if ($_SESSION['user_type'] == 'user') {
 
             header("Location: ../client/index.php");
 
+	    die();
+
         }
     }
 
@@ -250,14 +256,16 @@
         if (!check_user_login($_SESSION['user_logged'], $_SESSION['user_type'], $_SESSION['user_id'])) {
 
             header("Location: ../index.php");
+			die();
 
         }
 
     } else {
 
         header("Location: ../index.php");
-
+		die();
     }
+}
 
 function change_user_interface($form_id, $to_id) {
 
@@ -433,7 +441,7 @@
 
     write_log($from_udata['admin_name']." change into interface from ".$to_udata['admin_name']);
     return $header;
-	}
+
 }
 
 function unset_user_login_data () {
diff -Naur a/gui/reseller/new_ticket.php b/gui/reseller/new_ticket.php
--- a/gui/reseller/new_ticket.php	2005-12-31 12:11:01.000000000 +0100
+++ b/gui/reseller/new_ticket.php	2008-03-27 14:45:50.000000000 +0100
@@ -64,7 +64,7 @@
     $urgency = $_POST['urgency'];
 
     $subj = $_POST['subj'];
-
+    htmlspecialchars($_POST["user_message"], ENT_QUOTES, "UTF-8");
     $user_message = preg_replace("/\n/", "<br>", $_POST["user_message"]);
 
 	$ticket_status = 2;
@@ -96,7 +96,7 @@
                                          $urgency,
                                          $ticket_date,
                                          htmlspecialchars($subj, ENT_QUOTES, "UTF-8"),
-                                         htmlspecialchars($user_message, ENT_QUOTES, "UTF-8")));
+                                         $user_message));
 
 	send_tickets_msg($user_created_by,$user_id,$subj);
     header("Location: support_system.php");
diff -Naur a/gui/reseller/view_ticket.php b/gui/reseller/view_ticket.php
--- a/gui/reseller/view_ticket.php	2005-12-31 12:11:03.000000000 +0100
+++ b/gui/reseller/view_ticket.php	2008-03-27 14:46:11.000000000 +0100
@@ -273,7 +273,7 @@
     $ticket_date = time();
 
     $subj = $_POST['subject'];
-
+    htmlspecialchars($_POST["user_message"], ENT_QUOTES, "UTF-8");
     $user_message = preg_replace("/\n/", "<br>", $_POST["user_message"]);
 
 	$ticket_status = 2;
@@ -336,7 +336,7 @@
                                          $urgency,
                                          $ticket_date,
                                          htmlspecialchars($subj, ENT_QUOTES, "UTF-8"),
-                                         htmlspecialchars($user_message, ENT_QUOTES, "UTF-8")));
+                                         $user_message));
 
 	set_page_message(tr('Message was sent.'));
 	send_tickets_msg($ticket_from, $ticket_to, $subj);
diff -Naur a/gui/tools/webmail/inc/inc.php b/gui/tools/webmail/inc/inc.php
--- a/gui/tools/webmail/inc/inc.php	2005-12-31 12:01:55.000000000 +0100
+++ b/gui/tools/webmail/inc/inc.php	2007-12-10 23:20:49.000000000 +0100
@@ -150,9 +150,9 @@
 Don't remove the fallowing lines, or you will be problems with browser's cache 
 */
 
-Header("Expires: Wed, 11 Nov 1998 11:11:11 GMT\r\n".
-"Cache-Control: no-cache\r\n".
-"Cache-Control: must-revalidate");
+Header("Expires: Wed, 11 Nov 1998 11:11:11 GMT");
+Header("Cache-Control: no-cache");
+Header("Cache-Control: must-revalidate");
 
 $nocache = "
 <META HTTP-EQUIV=\"Cache-Control\" CONTENT=\"no-cache\">
